twake/backend/node/src/services/previews/services/processing/image.ts

```diff
@@ -0,0 +1,58 @@
+import sharp from "sharp";
+import { unlink } from "fs/promises";
+import { getTmpFile } from "../../utils";
+import { PreviewPubsubRequest, ThumbnailResult } from "../../types";
+
+export async function generatePreview(
+  inputPaths: string[],
+  options: PreviewPubsubRequest["output"],
+): Promise<{
+  output: ThumbnailResult[];
+  done: boolean;
+}> {
+  let output: ThumbnailResult[] = [];
+
+  for (const inputPath of inputPaths) {
+    var result: sharp.OutputInfo;
+
+    try {
+      const outputPath = getTmpFile();
+      const inputMetadata = await sharp(inputPath).metadata();
+      const outputFormat = computeNewFormat(inputMetadata, options);
+
+      result = await sharp(inputPath).resize(outputFormat).toFile(outputPath);
+      output.push({
+        path: outputPath,
+        width: result.width,
+        height: result.height,
+        type: "image/png",
+        size: result.size,
+      });
+    } catch (error) {
+      console.error("there was an error:", error.message);
+      return { output: [], done: false };
```

Comment:
sending back an optional error message will be nice if you return seomthing even when there is a failure